### ISSUE TITLE
Cannot login on macOS when a proxy is required for Internet access

### ISSUE THREAD

**Issue by cutiechi** [tailscale/tailscale#4263]
### What is the issue?

Can not login on latest macOS

### OS

macOS

### OS version

12.3

### Tailscale version

1.22.1

### Bug report

BUG-793761abc2b5d3a70cf14f1be6b95026bb8374c25157c20d8007ff2b1a028094-20220323071131Z-186ffe891a882114

**Comment by cutiechi** [comment]
Log:
<img width="694" alt="图片" src="https://user-images.githubusercontent.com/27606234/159644202-f2ee6d65-83c9-4ed9-805f-b43acfac8687.png">

**Comment by bradfitz** [comment]
Pointing my phone at that image, the text after "Signature check failed" translates as:

"The code fails to meet the specified code requirements"

Not sure what that means.

/cc @nickoneill

**Comment by cutiechi** [comment]
I already know what the reason is that proxy, how to set tailscale under proxy, example surge?

**Comment by bradfitz** [comment]
Sorry, I don't think we'll be able to help.

**Comment by cutiechi** [comment]
> Sorry, I don't think we'll be able to help.
> 
> 

I mean in my company there is no way to connect directly to your server and need to go through a proxy to connect so I want to ask how to configure your software behind my proxy

**Comment by JokerQyou** [comment]
Used to work well with proxy but the latest update broke it, it never finishes connecting now. Too bad can't use it at all. How do I delete my account?

**Comment by DentonGentry** [comment]
@JokerQyou https://login.tailscale.com/admin/settings/general has a delete button if that is what you want to do.

**Comment by DentonGentry** [comment]
Looking back at the original problem: /var/db/DetachedSignatures is only used if the system has to synthesize a code signature for an app that doesn’t have one. https://developer.apple.com/forums/thread/131498

It mentions network extension in the log output, so this is the App Store binary. It shouldn't be unsigned.

The App Store is not available in China. Was the app somehow sideloaded to get around this? That isn't going to work, macOS will only load network extensions when signed by the App Store. You might find https://formulae.brew.sh/formula/tailscale to be a more productive way to proceed.

**Comment by geckofu** [comment]
I'm trying to make Tailscale work behind a proxy as well, in my situation:

The proxy: [Surge](http://nssurge.com/) which is distributed outside of App Store

Tailscale:
- App Store version 1.26.0, installed using a US region Apple account (not sideloaded): works ✅ 
- Standalone version 1.26.0: not working, it hangs on the 'VPN starting' screen and never progress to the login step <sup>1</sup> 
- Homebrew version: can't have exit node working, the default route is not pointed to the Tailscale tunnel <sup>2</sup>

I'm interested in making either the standalone or the homebrew version working to distribute the app in our China office. 
Appreciate if you could provide guidelines on how to debug the problem. 

---

1. Standalone app console messages, code signing internal problem 🤔 
<img width="2372" alt="image" src="https://user-images.githubusercontent.com/4544326/173802239-cd7fc53d-1552-4861-8392-f01c0e1041f3.png">

2. Route table after I run `sudo tailscale up --exit-node=<IP>`
```
netstat -rn
Routing tables

Internet:
Destination        Gateway            RT_IFA             Flags        Refs      Use    Mtu          Netif Expire
default            192.168.124.1 (my router IP)     192.168.124.14     UGScg         147        0   1500            en0
100.64.183.255/32  utun3              100.69.119.66      USc             0        0   1280          utun3
...
```

**Comment by DentonGentry** [comment]
> I have tested the newly released v1.26.2 standalone app and it is working with my Surge proxy. 

I think this means the issue is resolved. Barring further comments, we'll expect to close this during the next regular bug scrub.

**Comment by cutiechi** [comment]
it's OK now
